OutOfSchool/OutOfSchool.WebApi.Tests/Services/ChildServiceTests.cs

```diff
@@ -260,37 +260,81 @@ public async Task GetApprovedByWorkshopId_NoExistingChildren_ShouldReturnEmptySe
-    //        };
-    //    }
+        // Act
+        await childService.DeleteChildCheckingItsUserIdProperty(child.Id, child.Parent.UserId, true);
 
-    //    private IEnumerable<Child> GetTestChildEntities()
-    //    {
-    //        return new List<Child>()
-    //        {
-    //            new Child { Id = 1, FirstName = "fn1", LastName = "ln1", MiddleName = "mn1", DateOfBirth = new DateTime(2003, 11, 9), Gender = Gender.Male, ParentId = 1, SocialGroupId = 2 },
-    //            new Child { Id = 2, FirstName = "fn2", LastName = "ln2", MiddleName = "mn2", DateOfBirth = new DateTime(2004, 11, 8), Gender = Gender.Female, ParentId = 2, SocialGroupId = 1 },
-    //            new Child { Id = 3, FirstName = "fn3", LastName = "ln3", MiddleName = "mn3", DateOfBirth = new DateTime(2006, 11, 2), Gender = Gender.Male, ParentId = 1, SocialGroupId = 1 },
-    //        };
-    //    }
-}
\ No newline at end of file
+        // Assert
+        applicationRepositoryMock.Verify(x => x.DeleteChildApplications(child.Id), Times.Once);
+        childRepositoryMock.Verify(x => x.Delete(It.Is<Child>(c => c.Id == child.Id)), Times.Once);
+    }
+
+    [Test]
+    public async Task ChildService_DeleteWithParentRole_DeletesChild()
+    {
+        // Arrange
+        var child = new Child()
+        {
+            Id = Guid.NewGuid(),
+            Parent = new Parent() { UserId = Guid.NewGuid().ToString() },
+        };
+
+        var childList = new List<Child> { child }.BuildMock();
+
+        childRepositoryMock.Setup(m => m.GetByFilterNoTracking(It.IsAny<Expression<Func<Child, bool>>>(), It.IsAny<string>()))
+            .Returns(childList);
+
+        childRepositoryMock.Setup(m => m.Delete(It.IsAny<Child>())).Returns(Task.CompletedTask);
+        applicationRepositoryMock.Setup(x => x.DeleteChildApplications(It.IsAny<Guid>())).Returns(Task.CompletedTask);
+
+        // Act
+        await childService.DeleteChildCheckingItsUserIdProperty(child.Id, child.Parent.UserId, false);
+
+        // Assert
+        applicationRepositoryMock.Verify(x => x.DeleteChildApplications(child.Id), Times.Once);
+        childRepositoryMock.Verify(x => x.Delete(It.Is<Child>(c => c.Id == child.Id)), Times.Once);
+    }
+
+    [Test]
+    public void ChildService_DeleteWithoutTechAdminPermissionAndInvalidUserId_ShouldNotDeleteChild()
+    {
+        // Arrange
+        var child = new Child()
+        {
+            Id = Guid.NewGuid(),
+            Parent = new Parent() { UserId = Guid.NewGuid().ToString() },
+        };
+
+        var childList = new List<Child> { child }.BuildMock();
+
+        childRepositoryMock.Setup(m => m.GetByFilterNoTracking(It.IsAny<Expression<Func<Child, bool>>>(), It.IsAny<string>()))
+            .Returns(childList);
+
+        childRepositoryMock.Setup(m => m.Delete(It.IsAny<Child>())).Returns(Task.CompletedTask);
+        applicationRepositoryMock.Setup(x => x.DeleteChildApplications(It.IsAny<Guid>())).Returns(Task.CompletedTask);
```

Comment:
(Test code excessiveness) Not sure that all Setups are required, as part of them (at least 2, if you assert section is correct) would never be called.

Reply:
Fixed

Reply:
I still insist, that neither `childRepository.Delete()` nor `applicationRepository.DeleteChildApplications()` should be setup in respective mocks. Moreover, you could check it if you add `.VerifyAll()` to `assert` section of the test.